src/widget/wwidgetstack.cpp

```diff
@@ -85,13 +96,32 @@ void WWidgetStack::hideIndex(int index) {
     }
 }
 
+void WWidgetStack::showEvent(QShowEvent*) {
+    int index = static_cast<int>(m_currentPageControl.get());
+    setCurrentIndex(index);
+
+    // Set the page triggers to match the current index.
+    for (QMap<int, ControlObject*>::iterator it = m_triggers.begin();
+            it != m_triggers.end(); ++it) {
+        it.value()->set(it.key() == index ? 1.0 : 0.0);
```

Comment:
Is this loop redundant, we should get et lest a show event for the current widget right? 
But anyway, just check how it is and leaf a comment about the signals we expect.